lib/src/participant/local_participant.dart

```diff
@@ -33,55 +25,176 @@ class LocalParticipant extends Participant {
       return Future.error(TrackPublishError('track already exists'));
     }
 
-    try {
-      final trackInfo =
-          await _engine.addTrack(cid: track.getCid(), name: track.name, kind: track.kind);
-      final transceiverInit = RTCRtpTransceiverInit(
-        direction: TransceiverDirection.SendOnly,
-      );
-      // addTransceiver cannot pass in a kind parameter due to a bug in flutter-webrtc (web)
-      track.transceiver = await _engine.publisher?.pc.addTransceiver(
-        track: track.mediaTrack,
-        init: transceiverInit,
-      );
-
-      final pub = LocalTrackPublication(trackInfo, track, this);
-      addTrackPublication(pub);
-      notifyListeners();
-
-      return pub;
-    } catch (e) {
-      return Future.error(e);
+    // try {
```

Comment:
while running in a debugger.. I've seen cases where uncaught exceptions in an async function was getting flagged. after doing some research, I couldn't figure out what Flutter's expectations were. Do you know if that's right?

Reply:
Looks good to me but I will need to do more real testing, I found another bug that needs to be fixed also.